### PR TITLE
Add tab autocompletion for device names

### PR DESCRIPTION
**Pull request by wkalt**

### CODE REVIEW

File: foxglove/cmd/root.go [hunk 1]
```diff
@@ -63,6 +63,26 @@ func listDevicesAutocompletionFunc(
...
 }
 
+func listDevicesByNameAutocompletionFunc(
+	baseURL string,
+	clientID string,
+	token string,
+	userAgent string,
+) func(*cobra.Command, []string, string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		client := console.NewRemoteFoxgloveClient(baseURL, clientID, token, userAgent)
```
Comment (bryfox) [inline foxglove/cmd/root.go:73]: I suspect this isn't picking up baseURL as expected, and fails silently in this case. Not a big deal, but if this is the case I wonder if we should skip the autocomplete if baseURL is specified.

**Review by bryfox — APPROVED** [review]
Completion needs to be re-run after this update? We should make a note of that in the readme, and in release notes if possible.

File: foxglove/cmd/root.go (hunk 1 above)
Comment (wkalt) [reply foxglove/cmd/root.go:73]: hmm, good point. You're probably right. I'll play around with this.

Comment (wkalt) [reply foxglove/cmd/root.go:84]: resolved with the approach discussed in slack

**Pull request merged**